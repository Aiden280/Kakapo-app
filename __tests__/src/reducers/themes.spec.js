import { prop } from 'ramda';
import { themeActions } from 'actions/';
import configureStore from 'stores/configureStore';

const store = configureStore();

const currState = () => store.getState().themes;

test('[reducer/themes]', () => {
  expect.assertions(7);
  store.dispatch(themeActions.change('#E91E63', 0));
  expect(prop('primary', currState())).toBe('#E91E63');
  expect(prop('btn', currState())).toBe('#4CAF50');

  store.dispatch(themeActions.change('#03A9F4', 0));
  expect(prop('primary', currState())).toBe('#03A9F4');

  store.dispatch(themeActions.change('#9C27B0', 1));
  expect(prop('btn', currState())).toBe('#9C27B0');

  store.dispatch(themeActions.change('#8BC34A', 1));
  expect(prop('btn', currState())).toBe('#8BC34A');

  store.dispatch(themeActions.change('#FFEB3B', 0));
  expect(prop('primary', currState())).toBe('#FFEB3B');
  expect(prop('darkUI', currState())).toBe(true);
});
