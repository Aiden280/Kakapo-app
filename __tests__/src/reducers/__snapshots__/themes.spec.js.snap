// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[reducer/themes] 1`] = `
Object {
  "btn": "#4CAF50",
  "colorPickerActive": false,
  "darkPrimary": "hsl(339.6, 82.2%, 41.3%)",
  "darkUI": false,
  "primary": "#E91E63",
  "version": "1.0.0",
}
`;

exports[`[reducer/themes] 2`] = `
Object {
  "btn": "#4CAF50",
  "colorPickerActive": false,
  "darkPrimary": "hsl(198.70000000000005, 97.6%, 38.7%)",
  "darkUI": false,
  "primary": "#03A9F4",
  "version": "1.0.0",
}
`;

exports[`[reducer/themes] 3`] = `
Object {
  "btn": "#9C27B0",
  "colorPickerActive": false,
  "darkPrimary": "hsl(198.70000000000005, 97.6%, 38.7%)",
  "darkUI": false,
  "primary": "#03A9F4",
  "version": "1.0.0",
}
`;

exports[`[reducer/themes] 4`] = `
Object {
  "btn": "#8BC34A",
  "colorPickerActive": false,
  "darkPrimary": "hsl(198.70000000000005, 97.6%, 38.7%)",
  "darkUI": false,
  "primary": "#03A9F4",
  "version": "1.0.0",
}
`;

exports[`[reducer/themes] 5`] = `
Object {
  "btn": "#8BC34A",
  "colorPickerActive": false,
  "darkPrimary": "hsl(53.89999999999998, 100%, 49.3%)",
  "darkUI": true,
  "primary": "#FFEB3B",
  "version": "1.0.0",
}
`;
